hack/generated/Makefile

```diff
@@ -70,15 +63,60 @@ generate: $(CONTROLLER_GEN) $(CONVERSION_GEN) ## Generate code
 #		--go-header-file=../boilerplate.go.txt
 
 	# Force regeneration of all of the CRDs
-	@echo "Deleting old CRDs"
+	@echo "Deleting old CRD YAMLs"
 	$(Q) if [ -d "./config/crd/bases" ]; then find "./config/crd/bases" -type f -name "*" -delete; fi
 
-	@echo "Executing controller-gen to generate CRDs"
-	$(Q) $(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./apis/..." output:crd:artifacts:config=config/crd/bases
+	@echo "Executing controller-gen to generate CRD and RBAC YAMLs"
+	$(Q) $(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
 
 .PHONY: build
 build: tidy lint ; $(info $(M) building ./bin/$(APP))
-	$(Q) $(GO) build -o ./bin/$(APP)
+	$(Q) CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on $(GO) build -o ./bin/$(APP)
+
+## --------------------------------------
+## Test
+## --------------------------------------
+.PHONY: test
+test: ; $(info $(M) running go test…)
+	$(Q) $(GO) test $(GO_DIRS_TO_TEST) -short -tags=noexit
+
+.PHONY: test-cover
+test-cover: $(GCOV2LCOV) ; $(info $(M) running go test…)
+	# NOTE: if you update the 'test-cover' target, also update ./github/workflows/test.yml
+	#       for the Windows part of the "test-generator" job.
+	$(Q) $(GO) test -short -tags=noexit -race -covermode atomic -coverprofile=cover.out -coverpkg=./... $(GO_DIRS_TO_TEST)
+	$(Q) $(GCOV2LCOV) -infile cover.out -outfile coverage.lcov
+
+# Initially this target uses kind as there is no need to end to end test on Azure.
+# Eventually when we start supporting features like MSI that only work in Azure,
+# it's likely we'll need to make the integration tests (or at least some of them)
+# run in an AKS cluster.
+.PHONY: test-int-no-cleanup
+test-int-no-cleanup: ; $(info $(M) running controller integration test…)
+	# MUST be executed as single command, or env vars will not propagate to test execution
+	$(Q) $(GO) test $(GO_DIRS_TO_TEST) # TODO: better way to just run integration tests?
+
+.PHONY: test-int-no-cleanup-cover
+test-int-no-cleanup-cover: $(GCOV2LCOV) ; $(info $(M) running controller integration test…)
+	# MUST be executed as single command, or env vars will not propagate to test execution
+	$(Q) $(GO) test -race -covermode atomic -coverprofile=cover.out -coverpkg=./... $(GO_DIRS_TO_TEST)
+	$(Q) $(GCOV2LCOV) -infile cover.out -outfile coverage.lcov
+
+.PHONY: test-int
+test-int: kind-create deploy test-int-no-cleanup kind-delete cleanup-test-azure-resources
+
+.PHONY: test-int-cover
+test-int-cover: kind-create deploy test-int-no-cleanup-cover kind-delete cleanup-test-azure-resources
+
+# Cleanup resource groups created by tests -- this isn't strictly required as the tests
+# clean up after themselves, but doing it here anyway just to be doubly sure we don't leak
+# resources in cases where the test pass is terminated, panics, etc
+.PHONY: cleanup-test-azure-resources
```

Comment:
I just realized that this isn't going to work well for parallel CI runs - will need to revisit it

Reply:
Improved this to do time-based deletion where we won't delete resource groups that were leaked if they're newer than a day (this caters for parallel runs and anyway the cleanup make target is just for extra safety as the tests cleanup after themselves anyway.